tool/gravity/cli/ops.go

```diff
@@ -87,110 +88,155 @@ func appPackage(env *localenv.LocalEnvironment) error {
 }
 
 func uploadUpdate(env *localenv.LocalEnvironment, opsURL string) error {
-	// create local environment with gravity state dir because the environment
-	// provided above has upgrade tarball as a state dir
-	localStateDir, err := localenv.LocalGravityDir()
+	clusterOperator, err := env.SiteOperator()
 	if err != nil {
-		return trace.Wrap(err)
+		return trace.Wrap(err, "unable to access cluster.\n"+
+			"Use 'gravity status' to check the cluster state and make sure "+
+			"that the cluster DNS is working properly.")
 	}
 
-	defaultEnv, err := localenv.New(localStateDir)
+	cluster, err := clusterOperator.GetLocalSite()
 	if err != nil {
 		return trace.Wrap(err)
 	}
 
-	clusterOperator, err := defaultEnv.SiteOperator()
+	tarballEnv, err := localenv.NewTarballEnvironment(localenv.TarballEnvironmentArgs{
+		License: cluster.License.Raw,
```

Comment:
Why is "state-dir" that comes from the command's CLI flag not passed in here in the config? That flag is supposed to indicate the state dir of the unpack tarball so it would make sense to propagate it here, no?